pkg/parser/external/java_parser.go

```diff
@@ -34,56 +32,69 @@ func newJavaParser(typ string) *javaParser {
 	return &javaParser{typ: typ}
 }
 
-type parseResult struct {
-	Statements []string `json:"statements"`
-	Position   int      `json:"position"`
-	Error      string   `json:"error"`
+func getServerAddress() string {
+	ip := getEnv("SQLFLOW_EXTERNAL_PARSER_IP", "localhost")
+	port := getEnv("SQLFLOW_EXTERNAL_PARSER_PORT", "12345")
+	return fmt.Sprintf("%s:%s", ip, port)
 }
 
-func (p *javaParser) Dialect() string {
-	return "java"
-}
-
-func (p *javaParser) Parse(program string) ([]string, int, error) {
-	// cwd is used to store train scripts and save output models.
-	cwd, err := ioutil.TempDir("/tmp", "sqlflow")
-	if err != nil {
-		return nil, -1, err
+func isServerUp(address string) bool {
+	cmd := exec.Command("curl", "-v", address)
+	if err := cmd.Run(); err != nil {
+		return false
 	}
-	defer os.RemoveAll(cwd)
+	return true
+}
 
-	inputFile := filepath.Join(cwd, "input.sql")
-	outputFile := filepath.Join(cwd, "output.json")
-	if err := ioutil.WriteFile(inputFile, []byte(program), 0755); err != nil {
-		return nil, -1, err
+func startServerIfNotUp() error {
+	address := getServerAddress()
+	if isServerUp(address) {
+		return nil
 	}
 
-	// TODO(yi): It is very expensive to start a Java process.  It
-	// slows down SQLFlow server's QPS if for every parsing
-	// operation, we'd have to start a Java process.
+	port := getEnv("SQLFLOW_EXTERNAL_PARSER_PORT", "12345")
```

Comment:
I am afraid that 12345 is a magic number. Consider that in the future we want to add another gRPC server -- if we allow default port, we might accidentally set its default port to 12345 too and make a conflict.

A safe approach might be that the program `log.Fatal` if there is no valid value of `SQLFLOW_PARSER_SERVER_PORT`.

Reply:
Sure, I will remove all occurrences of "12345" and define `SQLFLOW_PARSER_SERVER_PORT` in the Dockerfile.